Initializer/Initialize.cs

```diff
@@ -77,71 +128,77 @@ private void _writeLanguagesToDataBase()
         {
             //Get data from file
             List<ParsedData> languageData = ParseData(GetDataFromFile(
-                _pathRoot + _createPath("", "Languages", ".xlsx")));
+                _pathRoot + _generatePath("", "Languages", ".xlsx")));
 
             //Write data to languages
-            List<Language> allLanguages = new List<Language>();
             for (int i = 0, j = 0; i < languageData.Count; i++)
             {
-                if (allLanguages.FirstOrDefault(l => l.FullName == languageData[i].Word) == null)
+                if (_allLanguages.FirstOrDefault(l => l.FullName == languageData[i].Word) == null)
                 {
-                    allLanguages.Add(new Language()
+                    _allLanguages.Add(new Language()
                     {
                         ShortName = languageData[i + j].Language,
                         FullName = languageData[i + j++].Word
                     });
                 }
             }
 
-            if (!_isData)
+            //Check whether there is such data in the database
+            List<string> existingLanguages = _manageAccessToEntity.Languages.GetAll().Select(l => l.FullName).ToList();
+            if (!_isData || new HashSet<string>(_allLanguages.Select(l => l.FullName)).SetEquals(existingLanguages))
```

Comment:
very overcomplicated method - please use String.Equals method, with "InvariantCultureIgnoreCase" option

Reply:
Again should be fixed in all similar places